### ISSUE TITLE
Very slow on Android

### ISSUE THREAD

**Issue by slauzinho** [henninghall/react-native-date-picker#98]
It seems the issue #12  is back.
This is still happening on Android `"react-native-date-picker": "^2.6.1"` when I have a `maximumDate` set. Doesn't happen on iOS or when I remove that prop.

` "react": "16.5.0"
"react-native": 0.59`

**Comment by henninghall** [comment]
Hi, thanks for reporting this. What value is maximumDate set to? Do you keep minimumDate unset?

**Comment by slauzinho** [comment]
Hey have set `maximumDate`to be 18ys ago and`minimumDate` to be 100ys ago. 
I can confirm that this only happens when I have `maximumDate` and `minimumDate` set. If I have one and not the other it works fine

**Comment by henninghall** [comment]
Which mode are you using, date or datetime?

**Comment by slauzinho** [comment]
I'm using date, here is the component:


```javascript
<DatePicker
     date={this.state.dateOfBirth}
     mode={'date'}
     textColor="#FFFFFF"
     maximumDate={maximumDate}
     minimumDate={minimumDate}
     onDateChange={dateOfBirth => this.setState({ dateOfBirth })}
/>
```

**Comment by henninghall** [comment]
Thanks, I will look into it!

**Comment by henninghall** [comment]
Released a fix for this in v2.7.0

**Comment by LaikaTheSpaceDog** [comment]
@henninghall we are experiencing this issue on the latest version (4.3.5). Again, this only seems to occur on Android when the `maximumDate` prop is passed to the `DatePicker`. If we remove `maximumDate`, it works normally.
